File: w09/d03/matthew_korporaal/mail-object/mail_object_app/spec/javascripts/package_spec.js
describe(".package", function () {

  describe("Properties", function () {

    beforeEach(function () {
      package = new Package();
    });

    it("should have a content array", function () {
      expect(package.content).toBeDefined();
    });

    it("should have a fragile boolean", function () {
      expect(package.fragile).toBeDefined();
    });

    it("should have a broken boolean", function () {
      expect(package.broken).toBeDefined();
    });
  });

  describe("Methods", function () {
    beforeEach(function () {
      package = new Package();
    });

    it("should have a #shake function", function () {
      expect(package.shake).toBeDefined();
    });

    it("should be 'broken', if 'fragile' and #shake is invoked", function () {
      package.fragile = true;
      package.shake();
      expect(package.broken).toBe(true);
    });
  });
});

File: w09/d01/jimmy_davis/mail_items/spec/package_spec.rb
require_relative 'spec_helper'
require_relative '../lib/mailitem'
require_relative '../lib/package'

describe Package do

  it "should have a ship to address" do
    package = Package.new
    expect(package.ship_to_address).to be_true
  end

  it "should have a content array" do
    package = Package.new
    expect(package.content.length).to eq(2)
  end

  it "should be labeled if fragile or not" do
    package = Package.new
    package.is_fragile = true
    expect(package.is_fragile).to be_true
  end

  it "should be tell if it is broken or not" do
    package  = Package.new
    expect(package.is_broken).to be_false
  end

  context "should be broken if it is fragile and shaken" do
    it "should break if shaken" do
      p1 = Package.new
      p1.is_fragile = true
      expect(p1.is_fragile).to be_true
      expect(p1.is_broken).to be_false
      p1.shake
      expect(p1.is_broken).to be_true
    end
    it "should not break if shaken" do
      p2 = Package.new
      expect(p2.is_fragile).to be_false
      expect(p2.is_broken).to be_false
      p2.shake
      expect(p2.is_broken).to be_false
    end
  end

end
